### DISCUSSION TITLE
Using Piranha CMS in a net core 8..0 Blazor Web App

### DISCUSSION THREAD

**Discussion by ralfbalzer** [PiranhaCMS/piranha.core#2036] [Q&A]
I have been trying to use Piranha in combination with a Blazor Web App in net core 8.0.
But I can't get the routing to work. Has anyone got any idea on how to achieve this?

**Comment by tidyui** [comment]
Could you elaborate a bit more on how your application works. The built in routing in Piranha is for server side rendered web applications, not a SPA or client app.

**Comment by ralfbalzer** [comment]
Hi @tidyui, I was trying to use Blazor Pages. but I have come to the conclusion, that this is impossible, as you can't have a dynamic "page" directive on a Blazor Page.
Therefore all the Urls would have to be mapped in code to a Blazor Page, meaning that every new Page would have to be configured (or at least the page directive). Something like this: `@page "/myPage"`
